components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -0,0 +1,428 @@
+	rc := CmdResult{newSSHUtil.getSSHConfig().hostIP, []string{}, "", nil}
+	if len(inputFiles) != 0 {
+		for inputFile, outputFile := range inputFiles {
+			err := newSSHUtil.copyFileToRemote(inputFile, outputFile, false)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+		}
+	}
+
+	output, err := newSSHUtil.connectAndExecuteCommandOnRemote(command, true)
+	if err != nil {
+		rc.Error = err
+		resultChan <- rc
+		return
+	}
+
+	if len(outputFiles) != 0 {
+		for _, file := range outputFiles {
+			outFile := newSSHUtil.getSSHConfig().hostIP + "_" + file
+			destFileName, err := newSSHUtil.copyFileFromRemote(file, outFile)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+			writer.Printf("File downloaded %s \n", destFileName)
+			rc.OutputFiles = append(rc.OutputFiles, destFileName)
+		}
+	}
+
+	if errorCheck {
+		err = checkResultOutputForError(output)
+		if err != nil {
+			rc.Error = err
+			resultChan <- rc
+			return
+		}
+	}
+
+	rc.Output = output
+	resultChan <- rc
+}
+
+// preCmdExecCheck will check and execute PreExec function
+func preCmdExecCheck(node *Cmd, sshUtil SSHUtil, infra *AutomteHAInfraDetails, remoteService string, timestamp string, writer *cli.Writer) ([]string, error) {
+	var nodeIps []string
+	var err error
+	if (node.PreExec) != nil {
+		err = node.PreExec(node.CmdInputs, sshUtil, infra, remoteService, timestamp, writer)
+		if err != nil {
+			return nodeIps, err
+		}
+	}
+
+	if remoteService == CONST_FRONTEND {
+		nodeIps, err = isFrontendIPs(node.CmdInputs.NodeIp, infra)
+	} else if remoteService == CONST_AUTOMATE {
+		nodeIps, err = isAutomateIPs(node.CmdInputs.Single, node.CmdInputs.NodeIp, infra)
+	} else if remoteService == CONST_CHEF_SERVER {
+		nodeIps, err = isChefserverIPs(node.CmdInputs.Single, node.CmdInputs.NodeIp, infra)
+	} else if remoteService == CONST_POSTGRESQL {
+		nodeIps, err = isPostgresqlIPs(node.CmdInputs.Single, node.CmdInputs.NodeIp, infra)
+	} else if remoteService == CONST_OPENSEARCH {
+		nodeIps, err = isOpensearchIPs(node.CmdInputs.Single, node.CmdInputs.NodeIp, infra)
+	}
+
+	if err != nil {
+		return nodeIps, err
+	}
+	if len(nodeIps) == 0 {
+		errMsg := fmt.Sprintf("No %s IPs are found", remoteService)
+		return nodeIps, errors.New(errMsg)
+	}
+	return nodeIps, nil
+}
+
+// getSshDetails will return the SSH details.
+func getSshDetails(infra *AutomteHAInfraDetails) *SSHConfig {
+	sshConfig := &SSHConfig{
+		sshUser:    infra.Outputs.SSHUser.Value,
+		sshPort:    infra.Outputs.SSHPort.Value,
+		sshKeyFile: infra.Outputs.SSHKeyFile.Value,
+	}
+	return sshConfig
+}
+
+// isFrontendIPs will return the Frontend Node Ips.
+func isFrontendIPs(ip string, infra *AutomteHAInfraDetails) ([]string, error) {
+	frontendNodes := append(infra.Outputs.AutomatePrivateIps.Value, infra.Outputs.ChefServerPrivateIps.Value...)
+	if ip != "" {
+		if isValidIP(ip, frontendNodes) {
+			return []string{ip}, nil
+		} else {
```

Comment:
We can avoid else as you are returning in if so a return statement outside if should suffice

Reply:
Done